context/app/static/js/components/detailPage/visualization/Visualization/Visualization.jsx

```diff
@@ -70,8 +71,18 @@ function Visualization({ vitData, uuid, hasNotebook, shouldDisplayHeader, should
     onCopyUrlWarning,
     onCopyUrlSnackbarOpen,
     setOnCopyUrlSnackbarOpen,
+    setVizNotebookId,
   } = useVisualizationStore(visualizationStoreSelector);
 
+  // Propagate UUID to the store if there is a notebook so we can display the download button when the visualization is expanded
+  useEffect(() => {
+    if (hasNotebook) {
+      setVizNotebookId(uuid);
+    } else {
+      setVizNotebookId(null);
```

Comment:
So I can better understand this probably is not necessary now, but would be necessary once client state isn't refreshed on navigation? Not suggesting we remove it.

Reply:
I was thinking of this as a way to clean up the state in case we ever have cases where there's multiple visualizations on the same page, but this probably wouldn't even accomplish that on second thought. I'll drop this line to avoid the unnecessary state update

Reply:
I've removed the null setter in lieu of rerunning it every time a visualization's expansion status changes - this should make it so that if we eventually have a page with multiple vitessce instances that have notebooks (publications currently don't have notebooks) the proper visualization's ID is used.